Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -94,4 +115,77 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
         this._object.y,
       ])
     );
+//  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
+  var result = [];
+  var center = [this._object.x + this._radius, this._object.y + this._radius];
+  var halfOfDiag = Math.sqrt(2) * this._radius;
+
+  this._manager.getAllObstaclesAround(this._object, this._radius, result);
+  var hitBoxes = result.map(function (item) {
+    return item.owner.hitBoxes;
+  });
+  hitBoxes.push(this._object.getHitBoxes());
+
+  var polygons = hitBoxes.flat(1);
+  var vertices = polygons.map(function (poly) {
+    return poly.vertices;
+  }).flat(1);
+
+  function _calculatePOI(angle) {
+    var minDist = halfOfDiag;
+    var minPOI = [];
+    for (var poly of polygons) {
+      var poi = gdjs.Polygon.raycastTest(
+        poly,
+        center[0],
+        center[1],
+        center[0] + halfOfDiag * Math.cos(angle),
+        center[1] + halfOfDiag * Math.sin(angle)
+      )
+
+      if (poi.closeSqDist <= minDist) {
+        minDist = poi.closeSqDist;
+        minPOI.length = 0;
+        minPOI.push.apply(minPOI, [poi.closeX, poi.closeY]);
```

Comment:
As explained in my other comment:

```
minPOI[0] = poi.closeX
minPOI[1] = poi.closeY
```

this allocates no memory at all, just assigning two integers in memory, which is way way faster than erasing an array (`minPOI.length = 0;`), creating a new array (`[poi.closeX, poi.closeY]`), calling a function (`push`) that will probably then manipulate your array.